packages/grafana-runtime/src/utils/DataSourceWithBackend.ts

```diff
@@ -197,14 +198,16 @@ class DataSourceWithBackend<
       });
     }
 
+    const headers: Record<string, string> = {};
+    headers[PluginRequestHeaders.PluginID] = Array.from(pluginIDs).join(', ');
+    headers[PluginRequestHeaders.DatasourceUID] = Array.from(dsUIDs).join(', ');
+
     let url = '/api/ds/query';
     if (hasExpr) {
+      headers[PluginRequestHeaders.FromExpression] = 'true';
       url += '?expression=true';
```

Comment:
if this is set now as a header, I don't think we need to use both the query param and the header. The only place I see that is using this is here:

https://github.com/grafana/grafana/blob/main/pkg/services/query/models.go#L59

So we could modify that to use the header. Am I missing some other usage @marefr ?

Reply:
@ryantxu given #58082. Doesn't look like the query string is is use, but maybe there's still some magic use case of it in a "future multi-tenant world"?

Reply:
I'm merging this, so a reminder to create a separate issue for this conversation